File: android/src/main/kotlin/com/f2fk/geofence_foreground_service/utils/CalculateCenter.kt
package com.f2fk.geofence_foreground_service.utils

import com.google.android.gms.maps.model.LatLng

fun calculateCenter(coordinates: List<LatLng>): LatLng {
    if (coordinates.size == 1) {
        return coordinates.first()
    }

    var sumLatitude = 0.0
    var sumLongitude = 0.0

    for (coordinate in coordinates) {
        sumLatitude += coordinate.latitude
        sumLongitude += coordinate.longitude
    }

    val centerLatitude = sumLatitude / coordinates.size
    val centerLongitude = sumLongitude / coordinates.size

    return LatLng(centerLatitude, centerLongitude)
}

File: ios/Classes/utils/CalculateCenter.swift
import Foundation
import CoreLocation

func calculateCenter(coordinates: [CLLocationCoordinate2D]) -> CLLocationCoordinate2D {
    if coordinates.count == 1 {
        return coordinates.first!
    }

    var sumLatitude: Double = 0.0
    var sumLongitude: Double = 0.0

    for coordinate in coordinates {
        sumLatitude += coordinate.latitude
        sumLongitude += coordinate.longitude
    }

    let centerLatitude = sumLatitude / Double(coordinates.count)
    let centerLongitude = sumLongitude / Double(coordinates.count)

    return CLLocationCoordinate2D(latitude: centerLatitude, longitude: centerLongitude)
}

